azurerm/resource_arm_storage_account.go

```diff
@@ -185,6 +185,15 @@ func resourceArmStorageAccount() *schema.Resource {
 							Elem:     &schema.Schema{Type: schema.TypeString},
 							Set:      schema.HashString,
 						},
+						"default_action": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(storage.DefaultActionAllow),
+								string(storage.DefaultActionDeny),
+							}, true),
+							Default: string(storage.DefaultActionAllow),
```

Comment:
from a security perspective this feels like a default that could trip people up - it could be better to make this Required and let users intentionally select it?